platforms/documentation/docs/src/docs/userguide/dep-man/declaring-depedencies/declaring_dependencies_basics.adoc

```diff
@@ -0,0 +1,221 @@
+
+====
+include::sample[dir="snippets/dependencyManagement/declaringDependencies-fileDependencies/kotlin",files="build.gradle.kts[tags=file-dependencies]"]
+include::sample[dir="snippets/dependencyManagement/declaringDependencies-fileDependencies/groovy",files="build.gradle[tags=file-dependencies]"]
+====
+
+In this example, each dependency explicitly specifies its location within the file system.
+Common methods for referencing these files include:
+
+- link:link:{groovyDslPath}/org.gradle.api.Project.html#org.gradle.api.Project:files(java.lang.Object++[]++)[`Project.files()`]: Accepts one or more file paths directly.
+- link:{javadocPath}/org/gradle/api/file/ProjectLayout.html#files-java.lang.Object++...++-[`ProjectLayout.files()`]: Accepts one or more file paths directly.
+- link:{groovyDslPath}/org.gradle.api.Project.html#org.gradle.api.Project:fileTree(java.lang.Object)[`Project.fileTree()`]: Defines a directory and includes or excludes specific file patterns.
+
+[NOTE]
+====
+The order of files in a `FileTree` is not guaranteed to be stable, even on the same computer.
+As a result, dependency configurations using `FileTree` may produce resolution results with varying order, which can impact the cacheability of tasks that use these results as inputs.
+To ensure more predictable and stable behavior, it is recommended to use the simpler `files` method where possible.
+====
+
+Alternatively, you can use a <<declaring_repositories_adv.adoc#sub:flat_dir_resolver,flat directory>> repository to specify the source directory for multiple file dependencies.
+
+Ideally, you should use Maven or Ivy repository with a local URL:
+
+[source,groovy]
+----
+repositories {
+    maven {
+        url 'file:///path/to/local/files' // Replace with your actual path
+    }
+}
+----
+
+For more details on working with file dependencies, refer to the link:{groovyDslPath}/org.gradle.api.artifacts.dsl.DependencyHandler.html[`DependencyHandler`] and `ProjectLayout` classes in the Gradle API documentation.
+
+File dependencies also allow you to add a set of files directly to a configuration without using a repository.
+
+To add files as dependencies, pass a <<working_with_files.adoc#sec:file_collections,file collection>> to the configuration:
+
+====
+include::sample[dir="snippets/artifacts/externalDependencies/kotlin",files="build.gradle.kts[tags=file-dependencies]"]
+include::sample[dir="snippets/artifacts/externalDependencies/groovy",files="build.gradle[tags=file-dependencies]"]
+====
+
+Note that file dependencies are not included in the published dependency descriptor for your project.
+However, they are available in transitive dependencies within the same build, meaning they can be used within the current build but not outside it.
+
+You can specify which tasks produce the files for a file dependency.
+This is useful when files are generated during the build process:
```

Comment:
❗ I think this needs stronger language.  If you don't do this, and depend on these files transitively from another project, the tasks to generate them might not run, and you might not get them in your consumer.

Reply:
Updated.